src/sql/src/func.rs

```diff
@@ -2395,7 +2395,11 @@ lazy_static! {
                 params!(ArrayAny, ArrayAny) => BinaryFunc::Gte, 1075;
             },
             "=" => Scalar {
-                params!(Numeric, Numeric) => BinaryFunc::Eq, 1752;
+                params!(Numeric, Numeric) => Operation::binary(|_ecx, lhs, rhs| {
```

Comment:
one last general note that this change would need to get propagated to `List`s everywhere, as well